subscriptions/subscription.go

```diff
@@ -278,3 +279,48 @@ func ExtractFullTopicRef(fTopicRef string) (string, string, error) {
 	return items[1], items[3], nil
 
 }
+
+// GetMaxAckID gets a list of ack ids and selects the maximum one
+func GetMaxAckID(ackIDs []string) (string, error) {
+	var max int64
+	var maxAckID string
+
+	for _, ackID := range ackIDs {
+		tokens := strings.Split(ackID, "/")
+		if len(tokens) != 4 {
+			return "", errors.New("invalid argument")
+		}
+		subTokens := strings.Split(tokens[3], ":")
+		num, err := strconv.ParseInt(subTokens[1], 10, 64)
+
+		if err != nil {
+			return "", errors.New("invalid argument")
+		}
+
+		if num >= max {
+			max = num
+			maxAckID = ackID
+		}
+
+	}
+
+	return maxAckID, nil
+
+}
+
+// GetOffsetFromAckID extracts an offset from an ackID
+func GetOffsetFromAckID(ackID string) (int64, error) {
+
+	tokens := strings.Split(ackID, "/")
+	if len(tokens) != 4 {
+		return int64(-1), errors.New("invalid argument")
+	}
+	subTokens := strings.Split(tokens[3], ":")
+	num, err := strconv.ParseInt(subTokens[1], 10, 64)
+
+	if err != nil {
+		return int64(-1), errors.New("invalid argument")
```

Comment:
I am a bit confused why you are doing this here. Shouldn't you just return the ```err``` at the end of the function? In [handler.go](https://github.com/ARGOeu/argo-messaging/pull/59/files#diff-31a4a7824225629df071a1cba8b785d5R213) you do try to handle the error.